app/src/main/kotlin/at/bitfire/davdroid/ui/DebugInfoActivity.kt

```diff
@@ -120,101 +167,220 @@ class DebugInfoActivity : AppCompatActivity() {
+                    ) {
+                        TextButton(
+                            onClick = model::generateZip,
+                            enabled = !zipProgress
+                        ) {
+                            Text(
+                                stringResource(R.string.debug_info_archive_share).uppercase()
+                            )
+                        }
+                    }
+                }
             }
-        }
-
-        model.zipFile.observe(this) { zipFile ->
-            if (zipFile != null) {
-                // ZIP file is ready
-                val builder = ShareCompat.IntentBuilder(this)
-                    .setSubject("${getString(R.string.app_name)} ${BuildConfig.VERSION_NAME} debug info")
-                    .setText(getString(R.string.debug_info_attached))
-                    .setType("*/*")     // application/zip won't show all apps that can manage binary files, like ShareViaHttp
-                    .setStream(
-                        FileProvider.getUriForFile(
-                            this,
-                            getString(R.string.authority_debug_provider),
-                            zipFile
+            modelCause?.let { cause ->
+                item {
+                    CardWithImage(
+                        image = painterResource(R.drawable.undraw_server_down),
+                        title = when (cause) {
+                            is HttpException -> stringResource(if (cause.code / 100 == 5) R.string.debug_info_server_error else R.string.debug_info_http_error)
+                            is DavException -> stringResource(R.string.debug_info_webdav_error)
+                            is IOException, is IOError -> stringResource(R.string.debug_info_io_error)
+                            else -> cause::class.java.simpleName
+                        },
+                        subtitle = cause.localizedMessage,
+                        message = stringResource(
+                            if (cause is HttpException)
+                                when {
+                                    cause.code == 403 -> R.string.debug_info_http_403_description
+                                    cause.code == 404 -> R.string.debug_info_http_404_description
+                                    cause.code / 100 == 5 -> R.string.debug_info_http_5xx_description
+                                    else -> R.string.debug_info_unexpected_error
+                                }
+                            else
+                                R.string.debug_info_unexpected_error
+                        ),
+                        icon = Icons.Rounded.Info,
+                        modifier = Modifier.padding(horizontal = 8.dp, vertical = 8.dp)
+                    ) {
+                        TextButton(
+                            enabled = debugInfo != null,
+                            onClick = { shareFile(debugInfo!!) }
```

Comment:
My bad – I suggested to use shareFile, but we don't want to share the debug info here, but view it directly on the device. I use that very often for debugging.